test/Slacklinker/Handler/WebhookSpec.hs

```diff
@@ -95,6 +117,24 @@ spec = do
         ~[Entity _ theLink] <- runDB $ selectList [LinkedMessageRepliedThreadId ==. rtId] []
         ~(Just (Entity channelId _)) <- runDB $ getBy $ UniqueJoinedChannel wsId msg.channel
 
+        liftIO $ do
+          -- This should name the message that triggered slacklinker
+          theLink.joinedChannelId `shouldBe` channelId
+          theLink.messageTs `shouldBe` msg.ts
+          theLink.threadTs `shouldBe` Nothing
+          theLink.sent `shouldBe` False
+    it "can deal with a bot link" \app -> do
+      runAppM app $ do
+        (wsId, teamId) <- createWorkspace
+        let (url, parts) = sampleUrl
+        msg <- doBotLink teamId ts1 url
+
+        -- FIXME: MonadFail instead of irrefutable pattern crimes
```

Comment:
I think it wouldn't be hard to add the `MonadFail` instance for `AppM` (so that we don't need this workaround) since it's defined with this same package.  I think you could just add this to `src/Slacklinker/App.hs`:

```haskell
instance MonadFail AppM where
    fail string = liftIO (fail string)
```

(I have not actually checked if that works)

Reply:
Done, and this actually revealed a test failure also